runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/FarmingContractManager.java

```diff
@@ -0,0 +1,322 @@
+	{
+		PatchPrediction patchPrediction = farmingTracker.predictPatch(patch);
+		if (contract != null &&
+			patch.getRegion().getRegionID() == FARMING_GUILD_REGION_ID &&
+			contract.getPatchImplementation() == patch.getImplementation() &&
+			patchPrediction != null &&
+			(summary == SummaryState.EMPTY &&
+				(patchPrediction.getProduce() == null || patchPrediction.getProduce() == Produce.WEEDS)
+				|| patchPrediction.getProduce().equals(contract)))
+		{
+			return true;
+		}
+		return false;
+	}
+
+	public void loadContractFromConfig()
+	{
+		contract = getStoredContract();
+		handleContractState();
+	}
+
+	public boolean updateData(WorldPoint loc)
+	{
+		SummaryState oldSummary = summary;
+
+		handleContractState();
+		if (loc.getRegionID() == FARMING_GUILD_REGION_ID)
+		{
+			handleGuildmasterJaneWidgetDialog();
+			handleInfoBox();
+		}
+		else
+		{
+			if (infoBox != null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+		}
+		return oldSummary != summary;
+	}
+
+	private void handleInfoBox()
+	{
+		if (infoBox != null)
+		{
+			if (contract == null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+			else if (infoBox.getContract() != contract)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getItemID()), plugin, contract, config, this);
+				infoBoxManager.addInfoBox(infoBox);
+			}
+		}
+		else if (contract != null)
+		{
+			infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getItemID()), plugin, contract, config, this);
+			infoBoxManager.addInfoBox(infoBox);
+		}
+	}
+
+	private void handleGuildmasterJaneWidgetDialog()
+	{
+		Widget npcDialog = client.getWidget(WidgetInfo.DIALOG_NPC_HEAD_MODEL);
+
+		if (npcDialog == null || npcDialog.getModelId() != GUILDMASTER_JANE_WIDGET_MODEL_ID)
+		{
+			return;
+		}
+
+		String dialogText = Text.removeTags(client.getWidget(WidgetInfo.DIALOG_NPC_TEXT).getText());
+
+		if (dialogText.equals(CONTRACT_REWARDED))
+		{
+			setContract(null);
+		}
+
+		Matcher matcher = CONTRACT_ASSIGN_PATTERN.matcher(dialogText);
+
+		if (!matcher.find())
+		{
+			return;
+		}
+
+		String name = matcher.group(1);
+
+		Produce farmingContract = Produce.getByContractName(name);
+
+		if (farmingContract == null)
+		{
+			return;
+		}
+
+		Produce currentFarmingContract = contract;
+
+		if (farmingContract == currentFarmingContract)
+		{
+			return;
+		}
+
+		setContract(farmingContract);
+	}
+
+	private void handleContractState()
+	{
+		if (contract == null)
+		{
+			summary = SummaryState.UNKNOWN;
+			return;
+		}
+
+		PatchImplementation patchImplementation = contract.getPatchImplementation();
+
+		boolean hasEmptyPatch = false;
+		for (FarmingPatch patch : farmingWorld.getFarmingGuildRegion().getPatches())
+		{
+			if (patch.getImplementation() != patchImplementation)
+			{
+				continue;
+			}
+
+			PatchPrediction prediction = farmingTracker.predictPatch(patch);
```

Comment:
generally prefer having get + guards like these so the guard is attached to the get, as opposed to having a newline separating them:
```
			PatchPrediction prediction = farmingTracker.predictPatch(patch);
			if (prediction == null)
			{
				continue;
			}

			Produce produce = prediction.getProduce();
```